python/orca/src/bigdl/orca/learn/tf2/spark_runner.py

```diff
@@ -191,9 +181,7 @@ def _handle_batch_size(self, config):
 
 
 def find_ip_and_port(pre_iter):
-    tc = BarrierTaskContext().get()
```

Comment:
If there is no synchronization when finding free ports, I think there is a small chance that different tasks in the same machine will find the same free port. For example,

Task A in machine 1, runs into `find_free_port` and binds port 1234 and releases the port when return.
Then Task B in machine 1, which might be a little slower than Task A, could try to find a free port after Task A's `find_free_port` returns and finds 1234 again. 

This could result in different TensorFlow instances binding the same ip:port.

Reply:
Yes, I changed back to use tc again.